src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -224,11 +224,11 @@ struct AccessControlEntryCodec
         return staging.Encode(aWriter, aTag);
     }
 
-    CHIP_ERROR Decode(TLV::TLVReader & aReader)
+    CHIP_ERROR Decode(TLV::TLVReader & aReader, const Optional<FabricIndex> & aFabricIndex)
```

Comment:
Hi Song, just curious, is this all that needs to happen for this file (access control cluster server)? How can I check how this actually works, to ensure it handles the case of:

- entire list comes in for write acl attribute
- we transcode entire list into Access Control entries and write/update them in AccessControl module

Does it somehow "skip" any non-fabric entries? (If so, I need to ensure the index we keep track of during iteration is correct...)

Reply:
IIUC, It does not skip the non-fabric entries, since the decoder knows nothing about how the acl entries are stored actually.

Reply:
I think @mrjerryjohns was saying we only do the fabric-filtered write, so on the client side, we only ever send the fabric-filtered list. I just don't know what's going to come out on this side, all entries, or just fabric-filtered. Ideally, just fabric-filtered. But you know what, it doesn't matter right now, when this PR goes in, I will be pulling it and running it and I'll make it work here either way, my module code can handle either.

Reply:
This will only give you the fabric filtered entries, not the entire list of entries across all fabrics. So you'll have to manage that somehow..

Reply:
This logic here needs to be fixed for this PR to work correctly. Since ACLs are the target for a lot of this logic, it would make sense to show-case that being fixed up in this PR itself.

Reply:
Added a commit for adding fabricIndex argument to `UpdateEntry` functions, @mlepage-google please verify if the modified logic makes sense.
(I will do another update after write chunking PR)

Reply:
Yes, I was going to make that small change (using the accessing fabric index when indexing into ACL attribute) after this went in, I wouldn't hold up this PR for it. The sooner this PR goes in, the sooner I can start integration running and making sure everything is working well and there are some improvements I would like to make which are easier with a running system.